File: problems/1642.py
class Solution:
    def furthestBuilding(self, heights: List[int], bricks: int, ladders: int) -> int:
        diffs = [heights[i+1] - heights[i] for i in range(len(heights)-1)]
        print(diffs)
        ans = 0
        for diff in diffs:
            if diff > 0:
                if diff <= bricks:
                    bricks -= diff
                elif ladders > 0:
                    ladders -= 1
                else:
                    break
            ans += 1
        print(diff, bricks, ladders, ans)
        return ans

File: problems/1642.ts
function furthestBuilding(heights: number[], bricks: number, ladders: number): number {
  const diffs: number[] = [];
  for (let i = 0; i < heights.length-1; i++) {
    diffs.push(heights[i+1] - heights[i]);
  }
  console.log(diffs);
  let ans = 0;
  for (const diff of diffs) {
    if (diff > 0) {
      if (diff <= bricks) {
        bricks -= diff;
      }
      else if (ladders > 0) {
        ladders -= 1;
      }
      else {
        break;
      }
    }
    ans += 1;
    console.log(diff, bricks, ladders, ans);
  }
  return ans;
};
